API.md

```diff
@@ -0,0 +1,33 @@
+## API usage examples
+
```

Comment:
This file is really useful. Two points:

* there are also lots of useful bits in the pull request description that could go in here. I don't want to have to go find the pull request every time.
* this file isn't very discoverable. Could it be part of the README, or at least linked to by the README?

Reply:
🙌  on both parts.

I'll improve the content and merge into README.

I guess I was unsure what level of "done" we are going for :)